workflows/rnaseq-ref-index/make_pre_mrna_fasta.R

```diff
@@ -93,45 +99,69 @@ if (!dir.exists(opt$annotation_output)) {
 }
 
 
-# extract GRanges object containing genomic coordinates of each 
-# annotated transcript - both spliced transcripts and intronic regions
+# extract GRanges object containing genomic coordinates of each annotated transcript
+# both spliced transcripts and intronic regions
 grl <- eisaR::getFeatureRanges(
   gtf = opt$gtf,
   featureType = c("spliced", "intron"), 
+  flankLength = 86L,
+  joinOverlappingIntrons = FALSE,
   verbose = TRUE
 )
 
 # load in primary genome
 genome <- Biostrings::readDNAStringSet(opt$genome)
 names(genome) <- stringr::word(names(genome), 1)
 
-# extract spliced and intron sequences
-# genomic regions defined above
-seqs <- GenomicFeatures::extractTranscriptSeqs(
+
+# add the levels and lengths
+seqlevels(grl) <- seqlevels(genome)
+seqlengths(grl) <- seqlengths(genome)
+
+# trim grl used for splici
+splici_grl <- trim(grl)
+
+# extract seqs from trimmed grl for splici index
+splici_seqs <- GenomicFeatures::extractTranscriptSeqs(
   x = genome, 
-  transcripts = grl
+  transcripts = splici_grl
 )
 
-# write spliced and intron sequences to fasta file
+# remove sequence duplicates
+splici_seqs <- unique(splici_seqs)
+splici_grl <- splici_grl[names(splici_seqs)]
+
+# write splici to fasta 
 Biostrings::writeXStringSet(
-  seqs, filepath = spliced_intron_fasta, compress = TRUE
+  splici_seqs, filepath = spliced_intron_fasta, compress = TRUE
 )
 
 # write the associated annotations to gtf 
 eisaR::exportToGtf(
-  grl, 
+  splici_grl, 
   filepath = spliced_intron_gtf
 )
 
-# create text file mapping transcript and intron identifiers to 
-# corresponding gene identifiers
-# first make Tx2Gene for all spliced and intron sequences
-full_tx2gene <- eisaR::getTx2Gene(
-  grl, filepath = spliced_intron_tx2gene
-)
+# create text file mapping transcript and intron identifiers to corresponding gene identifiers
+# make 2 column Tx2Gene for all spliced and intron sequences
+splici_tx2gene_df <- eisaR::getTx2Gene(splici_grl)
+
+# write out 2 column Tx2 gene mapping
+readr::write_tsv(splici_tx2gene_df, spliced_intron_tx2gene, col_names = FALSE)
+
+# make 3 column Tx2 gene needed for alevin-fry USA mode
+splici_tx2gene_df_3col <- splici_tx2gene_df  %>%
+  # add status column
+  # remove extra -I* on gene ID
+  mutate(gene_id = stringr::word(gene_id,1,sep = '-'),
+         #if transcript_id contains an added -I*, then it is usnpliced
+         status = ifelse(stringr::str_detect(transcript_id, '-'), 'U', 'S'))
```

Comment:
Do we want to make this comment more explicit in the code?
```suggestion
         #if transcript_id contains an added -I*, then it is usnpliced
         status = ifelse(stringr::str_detect(transcript_id, '-I'), 'U', 'S'))
```

Reply:
I thought about that too and assume that it doesn't make a difference since it has to have a -I or -I followed by a number to be considered unspliced, so that will still do the trick and will make the change.